### DISCUSSION TITLE
[ZK Stack] - <send proveBatch transaction before provers finish jobs>

### DISCUSSION THREAD

**Discussion by qiwenzhou** [zkSync-Community-Hub/zksync-developers#563] [ZK Stack]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.74.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

After starting the server service, when a batch is submitted, the transactions for ProveBatch and ExecuteBatch will be immediately generated and attempted to be sent in the eth_tx table (at this time, the calculation of the batch by the Prover has not yet ended), accompanied by a "replacement price" warning ending up "invalid quotient evaluation" in sepolia. I am quite certain that the use of nonce is correct. This is the scene.
To sum up, there are two questions: first, why are the transactions of provideBatch and executeBatch generated in advance? The second issue is the occupation of nonce.
<img width="543" alt="image" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/31309283/09165f5c-e7b6-45e8-993d-fe2f7e164adc">
proof is set to used instead of skipping.

### Expected Behavior

the transactions for ProveBatch and ExecuteBatch are sent after prover jobs done.

### Current Behavior

sending two wrong transactions.

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by qiwenzhou** [comment]
This is very confusing. The server worked well for several days and suddenly failed to schedule its eth txs.

**Reply by bxpana** [reply]
Apologies for the delay here. Are you still having issues with this?

**Reply by qiwenzhou** [reply]
I encountered this issue again just yesterday. I ran zk stack prover-setup and cd prover && ./setup.sh gpu, rollbacked and restarted. It is not happening for now and I don't now why . Is there a explanation for this bothering issue?

**Reply by bxpana** [reply]
Not sure, but I'm asking on our end to see what might have caused that

**Reply by bxpana** [reply]
Didn't get an update on what might have caused this. If you run into the issue again please open up another discussion. Thanks!
